docs/Displaying a User Interface/Popup Keyboards/index.md

```diff
@@ -177,10 +185,120 @@ KeyboardListener keyboardListener = new KeyboardListener() {
     public void onKeyboardDidSendEvent(KeyboardEvent event, String currentInputText) {
         // This is sent upon every event, such as keypresses, cancellations, and aborting
     }
+
+    @Override
+    public void onKeyboardDidUpdateInputMask(KeyboardEvent event) {
+        switch (event) {
+            case INPUT_KEY_MASK_ENABLED:
+                // The user enabled input key masking
+                break;
+            case INPUT_KEY_MASK_DISABLED:
+                // The user disabled input key masking
+                break;
+            default:
+                break;
+        }
+    }
 };
 ```
 !@
 
+@![iOS, android, javaSE, javaEE]
+### Configuring the Keyboard Properties
+You can change default keyboard properties by updating !@@![iOS]`sdlManager.screenManager.keyboardConfiguration`!@@![android, javaSE, javaEE]`sdlManager.getScreenManager().setKeyboardConfiguration()`!@@![iOS, android, javaSE, javaEE]. For example, you can set the keyboard layout to a `NUMERIC` keyboard and allow the user to mask the input characters.
+![Numeric Keyboard](assets/keyboard_numeric_masked.png)
+!@
+
+@![iOS, android, javaSE, javaEE]
+To do that, you can set keyboard properties as the following 
+!@
+
+@![android, javaSE, javaEE]
+```java
+KeyboardProperties keyboardConfiguration = new KeyboardProperties()
+        .setKeyboardLayout(KeyboardLayout.NUMERIC)
+        .setMaskInputCharacters(KeyboardInputMask.ENABLE_INPUT_KEY_MASK);
+
+sdlManager.getScreenManager().setKeyboardConfiguration(keyboardConfiguration);
+```
+!@
+
+@![iOS]
+todo add iOS implementation 
+!@
+
+@![iOS, android, javaSE, javaEE]
+Each keyboard layout has a number of keys that can be customized to more serve the app needs. For example, you can set two of the customizable keys in `QWERTY` layout to be "!" and "?"
+![Custom Keys](assets/keyboard_querty_custom_keys.png)
+!@
```

Comment:
```suggestion
@![iOS, android, javaSE, javaEE]
#### Custom Keys (RPC 7.1+)
Each keyboard layout has a number of keys that can be customized to your app's needs. For example, you could set two of the customizable keys in `QWERTY` layout to be "!" and "?" as seen in the image below. The available number and location of these custom keys is determined by the connected head unit.

![Custom Keys](assets/keyboard_querty_custom_keys.png)
!@
```

Reply:
Please then also add subsections for setting a `language`, `limitedCharacterList`, and `autoCompleteList`